apps/examples/testcase/le_tc/kernel/itc_sched.c

```diff
@@ -0,0 +1,211 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file itc_sched.c
+/// @brief Test Case Example for Sched API
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <stdio.h>
+#include <pthread.h>
+#include <tinyara/config.h>
+#include "tc_internal.h"
+#include <stdbool.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <errno.h>
+#include <sys/wait.h>
+#include <sched.h>
+
+#define WAIT_TIME 1
+#define ARRLEN 2
+#define THREAD_CNT 3
+
+static bool g_pthread_callback;
+
+/**
+* @fn                   :thread_verifytcb_cb
+* @description          :Function for itc_sched_verifytcb_p
+* @return               :void*
+*/
+static void *thread_verifytcb_cb(void *param)
+{
+	bool ret_chk = false;
+	struct tcb_s *st_tcb;
+	st_tcb = sched_self();
+	if (st_tcb == NULL) {
+		g_pthread_callback = false;
+		return NULL;
+	}
+	ret_chk = sched_verifytcb(st_tcb);
+	if (ret_chk == false) {
+		g_pthread_callback = false;
+		return NULL;
+	}
+
+	g_pthread_callback = true;
+	return NULL;
+}
+
+/**
+* @fn                   :thread_schedself_cb
+* @description          :Function for itc_sched_self_p
+* @return               :void*
+*/
+static void *thread_schedself_cb(void *param)
+{
+	struct tcb_s *st_tcbself;
+	struct tcb_s *st_tcbpid;
+	st_tcbpid = sched_self();
+	if (st_tcbpid == NULL) {
+		g_pthread_callback = false;
+		return NULL;
+	}
+
+	/* should return tcb for current process */
+	st_tcbself = sched_self();
+	if (st_tcbself == NULL || st_tcbself->pid != st_tcbpid->pid) {
+		g_pthread_callback = false;
+		return NULL;
+	}
+
+	g_pthread_callback = true;
+	return NULL;
+}
+
+/**
+* * @fn                   :thread_schedself_cb
+* @description          :Function for itc_sched_self_p
+* @return               :void*
+*/
+static void thread_cleanup(pthread_t*  array,int thread_cnt)
```

Comment:
Please remove extra space "pthread_t*  array" and remove extra * symbol from comment "+* * @fn "

Reply:
This comment has been fixed.